yktrace/src/hwt/mapper.rs

```diff
@@ -271,4 +126,60 @@ impl HWTMapper {
         }
         ret
     }
+
+    /// Obtain the next mapped block.
+    ///
+    /// `last_yielded` must be a reference to the last successfully mapped block returned by the
+    /// iterator (or `None` if this is the first time you are calling `next`).
+    pub fn next(
+        &mut self,
+        last_yielded: Option<&IRBlock>,
+    ) -> Option<Result<IRBlock, HWTracerError>> {
+        while self.buffered_irblocks.is_empty() {
+            let block = self.trace_iter.next();
+            if block.is_none() {
+                return None;
+            }
+            let block = block.unwrap();
```

Comment:
`unwrap` is unnecessary if `match` is used.

Reply:
Same as before: I don't want to indent the remainder of the (already heavily indented) function.

Reply:
Getting rid of `unwrap` (which is dangerous and bloats the generated code) 100% beats indentation worries.

Reply:
Two questions:
 - Why is it dangerous? `block` cannot be `None` at line 143.
 - Why does it bloat the code?

Reply:
`unwrap`s get detached from their surrounding code over time. IMHO every `unwrap` must come with a comment explaining why it's correct, because of this danger.

`unwrap` has to put a branch and carry around the necessary error debugging stuff -- even though *we* know the error can't happen. [Sometimes the compiler might be able to get rid of that, but we can't guarantee that.]

Reply:
Fixed in 3f6535d0c5dd64637ac2e9f40c0c968a64e05170